dev/com.ibm.ws.ejbcontainer.mdb/src/com/ibm/ws/ejbcontainer/mdb/internal/MessageEndpointFactoryImpl.java

```diff
@@ -415,4 +419,71 @@ public String getActivationSpecId() {
     public String getDestinationId() {
         return beanMetaData.ivMessageDestinationJndiName;
     }
+
+    //PausableComponent Methods
+
+    /*
+     * (non-Javadoc)
+     *
+     * @see com.ibm.ws.kernel.launch.service.PauseableComponent#getExtendedInfo()
+     */
+    @Override
+    public HashMap<String, String> getExtendedInfo() {
+        throw new UnsupportedOperationException();
+    }
+
+    /*
+     * (non-Javadoc)
+     *
+     * @see com.ibm.ws.kernel.launch.service.PauseableComponent#getName()
+     */
+    @Override
+    public String getName() {
+        return name;
+    }
+
+    /*
+     * (non-Javadoc)
+     *
+     * @see com.ibm.ws.kernel.launch.service.PauseableComponent#isPaused()
+     */
+    @Override
+    public boolean isPaused() {
+        return !isEndpointActive();
+    }
+
+    /*
+     * (non-Javadoc)
+     *
+     * @see com.ibm.ws.kernel.launch.service.PauseableComponent#pause()
+     */
+    @Override
+    public void pause() throws PauseableComponentException {
+        try {
+            deactivateEndpoint();
+        } catch (ResourceException re) {
+            throw new PauseableComponentException(re);
+        }
+
+    }
+
+    /*
+     * (non-Javadoc)
+     *
+     * @see com.ibm.ws.kernel.launch.service.PauseableComponent#resume()
+     */
+    @Override
+    public void resume() throws PauseableComponentException {
+        try {
+            activateEndpoint();
```

Comment:
We should check here to see if we are already active, and if so log a message (or just output trace) and exit rather than go through all the setup/activate steps, only to discover we are already active.